Source/Editor/DualityEditor/Forms/ObjectRefSelectionDialog.cs

```diff
@@ -0,0 +1,281 @@
+				set { this._name = value; }
+			}
+
+			private string _path;
+
+			public string Path
+			{
+				get { return this._path; }
+				set { this._path = value; }
+			}
+
+			private IContentRef _resourceReference;
+
+			public IContentRef ResourceReference
+			{
+				get { return this._resourceReference; }
+				set { this._resourceReference = value; }
+			}
+
+			private GameObject _gameObjectReference;
+
+			public GameObject GameObjectReference
+			{
+				get { return this._gameObjectReference; }
+				set { this._gameObjectReference = value; }
+			}
+
+			private Component _componentReference;
+
+			public Component ComponentReference
+			{
+				get { return this._componentReference; }
+				set { this._componentReference = value; }
+			}
+		}
+
+		public Color PathColor { get; set; }
+		public Type FilteredType { get; set; }
+
+		public string ResourcePath { get; set; }
+
+		public IContentRef ResourceReference { get; set; }
+		public GameObject GameObjectReference { get; set; }
+		public Component ComponentReference { get; set; }
+
+		private TreeModel Model { get; set; }
+
+		public ObjectRefSelectionDialog()
+		{
+			InitializeComponent();
+
+			this.Model = new TreeModel();
+
+			this.PathColor = Color.Gray;
+
+			this.objectReferenceListing.SelectionChanged += this.ResourceListingOnSelectedIndexChanged;
+			this.objectReferenceListing.DoubleClick += this.ResourceListingOnDoubleClick;
+
+			this.txtFilterInput.KeyDown += this.TxtFilterInputOnKeyDown;
+
+			this.nodeName.DrawText += this.NodeName_OnDrawText;
+			this.nodePath.DrawText += this.NodePath_OnDrawText;
+		}
+
+		private void TxtFilterInputOnKeyDown(object sender, KeyEventArgs keyEventArgs)
+		{
+			TreeNodeAdv tmp = null;
+
+			if (keyEventArgs.KeyCode == Keys.Down)
+			{
+				tmp = this.objectReferenceListing.SelectedNode.NextNode;
+			}
+			else if (keyEventArgs.KeyCode == Keys.Up)
+			{
+				tmp = this.objectReferenceListing.SelectedNode.PreviousNode;
+			}
+
+			if (tmp != null)
+			{
+				this.objectReferenceListing.SelectedNode = tmp;
+			}
+		}
+
+		private void NodePath_OnDrawText(object sender, DrawTextEventArgs drawTextEventArgs)
+		{
+			ReferenceNode node = drawTextEventArgs.Node.Tag as ReferenceNode;
+
+			if (node == null)
+			{
+				return;
+			}
+
+			drawTextEventArgs.TextColor = this.PathColor;
+		}
+
+		private void NodeName_OnDrawText(object sender, DrawTextEventArgs drawTextEventArgs)
+		{
+			ReferenceNode node = drawTextEventArgs.Node.Tag as ReferenceNode;
+
+			if (node == null)
+			{
+				return;
+			}
+
+			drawTextEventArgs.TextColor = this.objectReferenceListing.ForeColor;
+		}
+
+		protected override void OnShown(EventArgs e)
+		{
+			base.OnShown(e);
+
+			this.objectReferenceListing.ClearSelection();
+			this.objectReferenceListing.Model = this.Model;
+
+			this.objectReferenceListing.BeginUpdate();
+			this.Model.Nodes.Clear();
+			
+			if (this.FilteredType.IsSubclassOf(typeof(GameObject)) || this.FilteredType == typeof(GameObject))

```

Comment:
Suggestion: Use `typeof(GameObject).IsAssignableFrom(this.FilteredType)` to wrap this up in a single check. Also covers cases where you'd have an interface type instead of `GameObject`.